[inductor][test] in test_unbacked_symints, replace inductor's skipCUDAIf with common device type's skipcudaif

henrylhtsang opened the pull request:
Differential Revision: D61506212

Use `skipCUDAIf` from `torch.testing._internal.common_device_type` if we create the test class with `instantiate_device_type_tests`. 

`instantiate_device_type_tests` would make sure the class has attr device_type, which works with`skipCUDAIf` from `torch.testing._internal.common_device_type`.

Also skipping test_vertical_pointwise_reduction_fusion for cpu test class, since the test expects cuda.

FAILED [0.0026s] test/inductor/test_unbacked_symints.py::TestUnbackedSymintsCPU::test_vertical_pointwise_reduction_fusion_cpu - AttributeError: 'TestUnbackedSymintsCPU' object has no attribute 'device'

repro: 
```
CUDA_VISIBLE_DEVICES="" pytest test/inductor/test_unbacked_symints.py -k cpu -v 
```


cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @ipiszy @yf225 @chenyang78 @kadeng @muchulee8 @ColinPeppler @amjames @desertfire @chauhang

henrylhtsang commented:
@pytorchbot rebase main

henrylhtsang commented:
@pytorchbot rebase -b main

ColinPeppler commented on test/inductor/test_unbacked_symints.py:
```diff
@@ -194,6 +197,9 @@ def test_vertical_pointwise_reduction_fusion(self, device):
         torch._inductor.metrics.reset()
 
+        if device == "cpu":
+            raise unittest.SkipTest("This test requires cuda")
```
no cpu for this guy?

henrylhtsang replied:
> no cpu for this guy?

@ColinPeppler it would directly call `tensor.cuda()` below, so cpu test won't work.

On the other hand, I tried replacing `cuda()` with `to(device)`, and it would run, but failed at numerical differences...

ColinPeppler replied:
I see thanks for the explanation!

ColinPeppler commented on torch/testing/_internal/inductor_utils.py:
```diff
@@ -72,6 +72,10 @@ def skipDeviceIf(cond, msg, *, device):
     if cond:
         def decorate_fn(fn):
             def inner(self, *args, **kwargs):
+                assert hasattr(self, "device"), (
+                    "Expect the test class to have attribute device but not found. "
+                    "Consider using the skip device decorators in common_device_type.py" if hasattr(self, "device_type") else ""
```
assert seems too aggressive, maybe log instead?

henrylhtsang commented:
@pytorchbot rebase -b main

henrylhtsang commented:
@pytorchmergebot merge

henrylhtsang committed:
[inductor][test] in test_unbacked_symints, replace inductor's skipCUDAIf with common device type's skipcudaif (#133936)

Summary:
Pull Request resolved: https://github.com/pytorch/pytorch/pull/133936

Use `skipCUDAIf` from `torch.testing._internal.common_device_type` if we create the test class with `instantiate_device_type_tests`.

`instantiate_device_type_tests` would make sure the class has attr device_type, which works with`skipCUDAIf` from `torch.testing._internal.common_device_type`.

Also skipping test_vertical_pointwise_reduction_fusion for cpu test class, since the test expects cuda.

Test Plan:
FAILED [0.0026s] test/inductor/test_unbacked_symints.py::TestUnbackedSymintsCPU::test_vertical_pointwise_reduction_fusion_cpu - AttributeError: 'TestUnbackedSymintsCPU' object has no attribute 'device'

repro:
```
CUDA_VISIBLE_DEVICES="" pytest test/inductor/test_unbacked_symints.py -k cpu -v
```

Reviewed By: ColinPeppler

Differential Revision: D61506212

izaitsevfb commented:
@pytorchbot merge -i
